src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -464,7 +464,7 @@ int GetDeliveryAttempts(BasicDeliverEventArgs message, string messageIdKey)
                 return attempts;
             }
 
-            if (message.BasicProperties.Headers.TryGetValue("x-delivery-count", out var headerValue))
+            if (message.BasicProperties.Headers != null && message.BasicProperties.Headers.TryGetValue("x-delivery-count", out var headerValue))
             {
                 attempts = Convert.ToInt32(headerValue) + 1;
```

Comment:
Seeing how crucial this code can be, should we not use something like Int32.TryParse here as well?

Reply:
Makes sense to me 👍

Reply:
This header value comes from the broker, so I didn't think it would be necessary to protect against the broker sending an invalid value.

Reply:
Ah, in that case I see less value in checking this as well 👍

Reply:
I _will_ remind you of this conversation if the next bug is in that line ;-)